### PR TITLE
[CELEBORN-1277] Add celeborn.quota.enabled at Master and Client side to enable checking quota

### PR DESCRIPTION
**Pull request by AngersZhuuuu**
### What changes were proposed in this pull request?

Add `celeborn.quota.enabled` at Master and Client side to enable checking quota

### Why are the changes needed?

`celeborn.quota.enabled` should be added in Master and Client side to enable quota check for Celeborn Master and Client.

### Does this PR introduce _any_ user-facing change?

Add categories of `celeborn.quota,enabled` with `master` and `client`.

### How was this patch tested?

No.

### CODE REVIEW

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala [hunk 1]
```diff
@@ -4163,9 +4163,12 @@ object CelebornConf extends Logging {
   val QUOTA_ENABLED: ConfigEntry[Boolean] =
     buildConf("celeborn.quota.enabled")
       .categories("quota")
-      .doc("When true, before registering shuffle, LifecycleManager should check " +
-        "if current user have enough quota space, if cluster don't have enough " +
-        "quota space for current user, fallback to Spark's default shuffle")
+      .doc(
```
Comment (RexXiong) [inline common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:4166]: add master/client to the categories?

Comment (AngersZhuuuu) [reply common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:4166]: How about current?

**Comment by AngersZhuuuu** [comment]
ping @RexXiong PTAL

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala [hunk 2]
```diff
@@ -4162,17 +4162,20 @@ object CelebornConf extends Logging {
 
   val QUOTA_ENABLED: ConfigEntry[Boolean] =
     buildConf("celeborn.quota.enabled")
-      .categories("quota")
-      .doc("When true, before registering shuffle, LifecycleManager should check " +
-        "if current user have enough quota space, if cluster don't have enough " +
-        "quota space for current user, fallback to Spark's default shuffle")
+      .categories("quota", "master", "client")
+      .doc(
+        "When Master side set true, Celeborn master will enable QuotaManager checking quota." +
```
Comment (SteNicholas) [inline common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:4167]:
```suggestion
        "When Master side sets to true, the master will enable to check the quota via QuotaManager." +
```

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala [hunk 3]
```diff
@@ -4162,17 +4162,20 @@ object CelebornConf extends Logging {
...
+          "When client side set true, before registering shuffle, " +
```
Comment (SteNicholas) [inline common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:4168]:
```suggestion
          "When Client side sets to true, LifecycleManager will request Master side to check whether the current user has enough quota before registration of shuffle. " +
```

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala [hunk 4]
```diff
@@ -4162,17 +4162,20 @@ object CelebornConf extends Logging {
...
+          "LifecycleManager will request to Master checking if current user have enough quota space, " +
+          "if cluster don't have enough quota space for current user, " +
```
Comment (SteNicholas) [inline common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:4170]:
```suggestion
          "Fallback to the default shuffle service of Spark when Master side checks that there is no enough quota for current user." +
```

**Review by SteNicholas — APPROVED** [review]
LGTM. cc @RexXiong

File: master/src/main/scala/org/apache/celeborn/service/deploy/master/Master.scala [hunk 5]
```diff
@@ -1064,10 +1064,14 @@ private[celeborn] class Master(
   private[master] def handleCheckQuota(
       userIdentifier: UserIdentifier,
       context: RpcCallContext): Unit = {
-    val userResourceConsumption = handleResourceConsumption(userIdentifier)
-    val (isAvailable, reason) =
-      quotaManager.checkQuotaSpaceAvailable(userIdentifier, userResourceConsumption)
-    context.reply(CheckQuotaResponse(isAvailable, reason))
+    if (conf.quotaEnabled) {
```
Comment (RexXiong) [inline master/src/main/scala/org/apache/celeborn/service/deploy/master/Master.scala:1067]: IMO quotaEnabled should only control quotaManager check quota or not，Resource-related metrics should be independent.

**Review by RexXiong — APPROVED** [review]
LGTM, except a nit

**Comment by AngersZhuuuu** [comment]
Merge to main(v0.5.0).

File: master/src/main/scala/org/apache/celeborn/service/deploy/master/Master.scala (hunk 5 above)
Comment (AngersZhuuuu) [reply master/src/main/scala/org/apache/celeborn/service/deploy/master/Master.scala:1040]: Done

**Pull request closed without merging**